coreneuron/apps/main1.cpp

```diff
@@ -102,6 +103,11 @@ char* prepare_args(int& argc, char**& argv, int use_mpi, const char* arg) {
     args.append(" --skip-mpi-finalize ");
     if (use_mpi) {
         args.append(" --mpi ");
+    } else if(args.find(" --mpi ") != std::string::npos ) {
```

Comment:
@alexsavulescu : is `use_mpi` was false, I was not expecting to have `--mpi` as arg here. Did you see why it comes here?

Reply:
https://github.com/BlueBrain/CoreNeuron/issues/312 

from channel-benchmark:

```
strdef arg 
sprint(arg, "--tstop %lf --mpi", tstop)
pnm.pc.nrncore_run(arg)
```

Reply:
Ok. In that case I would change/drop this because we are removing `pnm.pc.nrncore_run` from user facing side. With https://github.com/neuronsimulator/nrn/pull/646, we do:

```
    from neuron import coreneuron
    coreneuron.enable = True
    pc = h.ParallelContext()
    h.stdinit()
    pc.psolve(h.tstop)
```

and neuron will take care of everything. We don't have to pass those ugly params manually.

Reply:
I see. Will close the PR then.